.gitignore

```diff
@@ -11,3 +11,6 @@ tests/*tests.dir
 Testing/Temporary/*.*
 CMakeCache.txt
 *.suo
+.vs/
+.vscode/
+build*/
```

Comment:
Ignoring `.vs/` and `.vscode` is all fine and dandy, but I did not intend to commit my "ignore all the build folders" line here. Shout if you'd like me to remove this from the PR.